src/iop/demosaic.c

```diff
@@ -606,7 +606,13 @@ static void xtrans_markesteijn_interpolate(float *out, const float *const in,
           if((col >= 0) && (row >= 0) && (col < width) && (row < height))
           {
             const int f = FCxtrans(row, col, roi_in, xtrans);
-            for(int c = 0; c < 3; c++) pix[c] = (c == f) ? in[roi_in->width * row + col] : 0;
+            for(int c = 0; c < 3; c++) pix[c] = (c == f) ? in[roi_in->width * row + col] : 0.f;
+            // There should be no negative values at this point in the
+            // pixel pipeline as this could only happen with negative
+            // sensor readings. (It is possible to have negative
```

Comment:
Good to know. I can update the comment to reflect this without removing the assert? I don't think it actually bothers Markesteijn demosaic to have input outside of [0,processed_maximum] but as it will be clamped to [0,processed_maximum] at end of this demosaic, seems good to lay out an expected initial condition.